### PR TITLE
My script implementation

### PR DESCRIPTION
**Pull request by danpaldev**
I put all the logic inside functions as your example. In this way the cases are more readable 😃

### CODE REVIEW

File: proc_info.sh [hunk 1]
```diff
@@ -14,4 +14,78 @@ function usage(){
   echo "-c                   Show command that started the process"
   echo "-f                   Show opened files by the process"
   echo "-h                   Show this help"
+  exit 0
```
Comment (korney4eg) [inline proc_info.sh]: what does this line?

File: proc_info.sh [hunk 2]
```diff
@@ -14,4 +14,78 @@ function usage(){
...
+      shift
+      if test $# -gt 0; then
+        get_opened_files $1
+      else
+        echo "no process specified"
+        exit 1
+      fi
+      shift
+      ;;
+    *) usage ;;
```
Comment (korney4eg) [inline proc_info.sh]: usually when you specify parameter that not exists, program should exit with non-zero return codes. 
For `usage` function it's ok to return `0` exit codes, all other cases should be marked as error.
Check out https://www.cyberciti.biz/faq/bash-get-exit-code-of-command/

File: proc_info.sh [hunk 3]
```diff
@@ -14,4 +14,78 @@ function usage(){
...
+}
+
+#Function for -f flag
+get_opened_files(){
+  pid=$1
+  ls -l /proc/$pid/fd | less | awk '{ print $11 }'
+}
+
+while test $# -gt 0; do
+  case "$1" in
```
Comment (korney4eg) [inline proc_info.sh]: Here you check first parameter and suppose that it should be flag like `-u`. In usage function it said that `PID` should go first.

File: proc_info.sh [hunk 4]
```diff
@@ -15,3 +15,76 @@ function usage(){
...
+get_command(){
+  pid=$1
+  #cat /proc/$pid/cmdline ; echo
+  IFS=''
+  cat /proc/$pid/cmdline |
+  while read data
+  do
+    echo "$data"
+  done
+}
```
Comment (korney4eg) [inline proc_info.sh:37]: After update I got this:
```
vagrant@wordpress:~$ sleep 1000 &
[3] 2141
vagrant@wordpress:~$ /vagrant/proc_info.sh 2141 -c
vagrant@wordpress:~$
```

Comment (danpaldev) [reply proc_info.sh:37]: I'm getting the same behavior! Gonna have to fix that, thanks for letting me know.

Comment (korney4eg) [reply proc_info.sh:37]: BTW. I'm testing it on VM created by Vagrant. So it's should be always same result.

**Comment by korney4eg** [comment]
I'm closing this PR as it does a job.

**Pull request closed without merging**